mantle/harness/harness.go

```diff
@@ -107,6 +108,14 @@ func (t *H) runTimeoutCheck(ctx context.Context, timeout time.Duration, f func()
 	}
 }
 
+// This functionn is robust to being called multiple times
+func (t *H) Release() {
+	if !t.released {
+		t.released = true
+		t.suite.release()
```

Comment:
I don't quite fully understand the behavior we are overcoming by only calling `t.suite.release()` per `H` versus what we are doing before this change. Can you help me understand? Maybe we need some comments?

Reply:
Before this change, `t.suite.release()` would be called once per `H` but the responsibility for that condition fell on `tRunner`. Now, we are exposing the releasing mechanism to all the platforms. So we can't trust `tRunner` to make that condition true, since there are also other actors that may "release" a test. Which is why we have added a wrapper here to ensure that a test is only released once. 

Note: if a test is released more than once, an inaccurate state is created in the queue

Reply:
> Before this change, `t.suite.release()` would be called once per `H` but the responsibility for that condition fell on `tRunner`. Now, we are exposing the releasing mechanism to all the platforms. So we can't trust `tRunner` to make that condition true, since there are also other actors that may "release" a test. Which is why we have added a wrapper here to ensure that a test is only released once.

Thanks. Maybe good to include that in the commit message too (maybe even break it into a separate first commit). 

> 
> Note: if a test is released more than once, an inaccurate state is created in the queue

So we are fixing a bug here?

Reply:
> Which is why we have added a wrapper here to ensure that a test is only released once.

This is https://en.wikipedia.org/wiki/Idempotence btw

Reply:
> So we are fixing a bug here?

Technically, there was no bug because a test would never be released more than once because only `tRunner` was releasing tests. With this change, its a good idea to add idempotence. For ex. an improper state in queue will be created if we try to release a test early via the new callback, and then `tRunner` also tries to release the test.